src/gs-category-page.c

```diff
@@ -230,6 +235,97 @@ copy_os_to_usb_button_cb (GtkButton *button, GsCategoryPage *self)
 			 self->cancellable);
 }
 
+static gchar*
+removable_destination_dup (GsCategoryPage *self)
+{
+	/* first one is the most-recently added mount on a removable disk */
+	if (self->copy_dests)
+		return g_strdup (self->copy_dests->data);
+
+	return NULL;
+}
+
+static void
+gs_category_page_os_get_copyable_cb (GObject *source_object,
+				     GAsyncResult *res,
+				     gpointer user_data)
+{
+	GsPluginLoader *plugin_loader = GS_PLUGIN_LOADER (source_object);
+	GsCategoryPage *self = GS_CATEGORY_PAGE (user_data);
+	g_autoptr(GError) error = NULL;
+	gboolean copyable;
+
+	copyable = gs_plugin_loader_job_os_get_copyable_finish (plugin_loader,
+								res, &error);
+	gtk_widget_set_visible (self->usb_action_box, copyable);
+}
+
+static void
+gs_category_page_update_os_copy_button (GsCategoryPage *self)
+{
+	g_autofree gchar* copy_dest = NULL;
+	g_autoptr(GsPluginJob) plugin_job = NULL;
+
+	if (self->category == NULL ||
+	    g_strcmp0 (gs_category_get_id (self->category), "usb") != 0) {
+		gtk_widget_set_visible (self->usb_action_box, FALSE);
+		return;
+	}
+
+	gtk_widget_set_visible (self->usb_action_box, TRUE);
+	gtk_widget_set_sensitive (self->copy_os_to_usb_button, FALSE);
+	gtk_widget_set_visible (self->copy_os_to_usb_button, FALSE);
+
+	if (self->copying)
+		return;
+
+	copy_dest = removable_destination_dup (self);
+	if (copy_dest != NULL) {
+		gtk_button_set_label (GTK_BUTTON (self->copy_os_to_usb_button), _("Copy OS to USB"));
+		gtk_widget_set_sensitive (self->copy_os_to_usb_button, TRUE);
+
+		/* also check (asynchronously) whether the OS will
+		 * certainly fail a copy so we can update the UI
+		 * accordingly. Cases where the plugin would know a copy
+		 * would fail include an OSTree remote which lacks a collection
+		 * ID or broader issues like the user lacking write
+		 * access to the destination. */
+		plugin_job = gs_plugin_job_newv (
+			GS_PLUGIN_ACTION_OS_GET_COPYABLE,
+			"failure-flags", GS_PLUGIN_FAILURE_FLAGS_NONE,
+			"copy-dest", copy_dest,
+			 NULL);
+		gs_plugin_loader_job_os_get_copyable_async (
+			self->plugin_loader,
+			plugin_job,
+			self->cancellable,
+			gs_category_page_os_get_copyable_cb,
+			self);
+	} else {
+		gtk_button_set_label (GTK_BUTTON (self->copy_os_to_usb_button), _("Insert USB to copy"));
```

Comment:
Nitpick: The GNOME style guide says that button labels should be in Title Case, i.e. `Insert USB to Copy`. A mnemonic would also be good.

Reply:
Should it also be `Insert USB to Copy To`?

Reply:
> Nitpick: The GNOME style guide says that button labels should be in Title Case, i.e. `Insert USB to Copy`. A mnemonic would also be good.

I'll file a follow-up task. We just got this string translated into a few languages and I assume you need a perfect match to take advantage of the translations.

> Should it also be `Insert USB to Copy To`?

Shouldn't it be `Insert USB to Which to Copy`? :)

The string I used has trivially different meaning but the point of both is the same. In that case, shorter wins. As it stands, the German translation is already long enough: "Einfügen von USB zum Kopieren". (I'm not sure the "von" is needed but Google Translate seems to be reasonably accurate for German and I get increasingly inaccurate with time).